File: C# Fundamentals/C# OOP Basics/07. Encapsulation/Ex/P02_ClassBoxDataV/Box.cs
using System;
using System.Collections.Generic;
using System.Text;

public class Box
{
    private double length;
    private double width;
    private double height;

    public double SurfaceArea()
    {
        //2lw + 2lh + 2wh
        return (2 * this.Length * this.Width) + (2 * this.Length * this.Height) + (2 * this.Width * this.Height);
    }

    public double LateralSurfaceArea()
    {
        //2lh + 2wh
        return (2 * this.Length * this.Height) + (2 * this.Width * this.Height);
    }

    public double Volume()
    {
        //lwh
        return this.Length * this.Width * this.Height;
    }

    public Box(double length, double width, double height)
    {
        this.Length = length;
        this.Width = width;
        this.Height = height;
    }

    public double Length
    {
        get
        {
            return this.length;
        }
        private set
        {
            if(value <= 0)
            {
                throw new ArgumentException("Length cannot be zero or negative.");
            }
            this.length = value;
        }
    }
    public double Width
    {
        get
        {
            return this.width;
        }
        private set
        {
            if (value <= 0)
            {
                throw new ArgumentException("Width cannot be zero or negative.");
            }
            this.width = value;       
        }
    }
    public double Height
    {
        get
        {
            return this.height;
        }
        private set
        {
            if (value <= 0)
            {
                throw new ArgumentException("Height cannot be zero or negative.");
            }
            this.height = value;           
        }
    }
}

File: Java/01.Advanced/02.OOP/07.Encapsulation/Ex/src/t_01_class_box/Box.java
package t_01_class_box;

public class Box {
    private double length;
    private double width;
    private double height;

    Box (double length, double width, double height){
        this.setLength(length);
        this.setWidth(width);
        this.setHeight(height);
    }

    public double calculateSurfaceArea () {
        return (2 * length * width) + (2 * length * height) + (2 * width * height);
    }

    public double calculateLateralSurfaceArea () {
        return (2 * length * height) + (2 * width * height);
    }

    public double calculateVolume () {
        return length * width * height;
    }


    public void setLength(double length) {
        if (length <= 0) {
            throw new IllegalArgumentException("Length cannot be zero or negative.");
        }

        this.length = length;
    }

    public void setWidth(double width) {
        if (width <= 0) {
            throw new IllegalArgumentException("Width cannot be zero or negative.");
        }

        this.width = width;
    }

    public void setHeight(double height) {
        if (height <= 0) {
            throw new IllegalArgumentException("Height cannot be zero or negative.");
        }

        this.height = height;
    }
}
